freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/ClusterProxyService.java

```diff
@@ -88,17 +102,27 @@
         LOGGER.debug("Registering freeipa with cluster-proxy: Environment CRN = [{}], Stack CRN = [{}]", stack.getEnvironmentCrn(), stack.getResourceCrn());
 
         GatewayConfig primaryGatewayConfig = gatewayConfigService.getPrimaryGatewayConfig(stack);
-        HttpClientConfig httpClientConfig = tlsSecurityService.buildTLSClientConfigForPrimaryGateway(
-                stack, primaryGatewayConfig.getGatewayUrl());
+        List<GatewayConfig> gatewayConfigs = gatewayConfigService.getNotTerminatedGatewayConfigs(stack);
         ClientCertificate clientCertificate = clientCertificates(stack);
 
-        ClusterServiceConfig serviceConfig = createServiceConfig(stack, httpClientConfig, clientCertificate);
+        boolean usePrivateIpToTls = stack.getSecurityConfig().isUsePrivateIpToTls();
+        List<GatewayConfig> tunnelGatewayConfigs;
+        List<ClusterServiceConfig> serviceConfigs = new LinkedList<>();
+        serviceConfigs.add(createServiceConfig(stack, FREEIPA_SERVICE_NAME, primaryGatewayConfig, clientCertificate, usePrivateIpToTls));
```

Comment:
is it ok to add this `ServiceConfig` on `bootstrap` and again when we are adding the other ones? It looks like to me we double add it, and I don't know if it cause any issue on CP side

Reply:
Yes, I confirmed this with the cluster proxy team. The registration is a "force" registration and it overwrites the old config.